### PR TITLE
cmd/kube-apiserver/app/options: Improving test coverage

### PR DESCRIPTION
**Pull request by tmilox**
Signed-off-by: TommyStarK <thomasmilox@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will increase the unit tests code coverage of `cmd/kube-apiserver/app/options`

- Initial unit tests code coverage:
Screenshot 2022-11-10 at 21 09 09

- After changes:
Screenshot 2022-11-10 at 21 09 40

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
```

### CODE REVIEW

**Comment by TommyStarK** [comment]
Hello @cheftako , @liggitt  😃

This is my very first contribution to Kubernetes,. In order to start taming the beast I thought that trying to increase unit tests code coverage of a small piece could be a good first step (I tried to look for a good first issue but they are all assigned).

Feel free to correct/guide me, I would really appreciate.

Best,

**Comment by TommyStarK** [comment]
@cheftako, @liggitt,

I know guys you are super busy and this PR is far from being a priority. I just wanted to know if this is acceptable as a first contribution. The intent was to test some edge cases not covered (I also took the opportunity to correct a typo in test description).

Best regards,

**Comment by alexzielenski** [comment]
/triage accepted

@cheftako please take a look if you have time, or delegate to someone else. thanks!

File: cmd/kube-apiserver/app/options/validation_test.go [hunk 1]
```diff
@@ -106,11 +113,6 @@ func TestClusterServiceIPRange(t *testing.T) {
 			expectErrors: false,
 			options:      makeOptionsWithCIDRs("10.0.0.0/16", "3000::/108"),
 		},
-		{
-			name:         "valid v6-v4 dual stack",
-			expectErrors: false,
-			options:      makeOptionsWithCIDRs("3000::/108", "10.0.0.0/16"),
-		},
```
Comment (liggitt) [inline cmd/kube-apiserver/app/options/validation_test.go:113]: why was this case dropped?

File: cmd/kube-apiserver/app/options/validation_test.go [hunk 2]
```diff
@@ -89,7 +92,11 @@ func TestClusterServiceIPRange(t *testing.T) {
 			expectErrors: true,
 			options:      makeOptionsWithCIDRs("10.0.0.0/16", "3000::/64"),
 		},
-
+		{
+			name:         "more than two entries",
+			expectErrors: true,
+			options:      makeOptionsWithCIDRs("10.0.0.0/16,244.0.0.0/16", "3000::/108"),
```
Comment (liggitt) [inline cmd/kube-apiserver/app/options/validation_test.go:98]: was the combination of two options into a single string intentional? this looks sort of weird

File: cmd/kube-apiserver/app/options/validation_test.go (hunk 1 above)
Comment (TommyStarK) [reply cmd/kube-apiserver/app/options/validation_test.go]: A mistake from my part, sorry. Resolved in 44c94f98b64843562323b1b9c12297be40023bdc

File: cmd/kube-apiserver/app/options/validation_test.go (hunk 2 above)
Comment (TommyStarK) [reply cmd/kube-apiserver/app/options/validation_test.go:98]: Yes it was intentional, the intent was to cover this condition: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/validation.go#L46

**Comment by TommyStarK** [comment]
Thank you very much for the review @liggitt. 
All checks have passed, don't hesitate if there is anything else I can do. 

Best,

**Pull request merged**